### PR TITLE
Renames Add-CloudEvent*Data to Set-CloudEvent*Data

### PR DESCRIPTION
**Pull request by dmilov**
* Addresses issue #9 renaming `Add-CloudEvent*Data` functions to `Set-CloudEvent*Data`. As described in the issue the Add notion is not correct for the functions functionality. `Set` verb notion matches the behavior of the functions.
* Increment minor version of the module. The module is not yet released but this is a backward compatibility breaking change. 
* Remove module catalog file creation from build.ps1 because catalog file is needed only when module is signed. We still don't have a signing implementation for this module.
* Testing done:
```powershell
> .\build.ps1
[11:24:08 AM] INFO: Publish CloudEvents.Sdk Module to 'C:\git-repos\dmilov-forks\sdk-powershell\CloudEvents.Sdk'
Microsoft (R) Build Engine version 16.8.3+39993bd9d for .NET
Copyright (C) Microsoft Corporation. All rights reserved.
                                                                                                                    Determining projects to restore...                                                                                All projects are up-to-date for restore.                                                                          CloudEventsPowerShell -> C:\git-repos\dmilov-forks\sdk-powershell\src\CloudEventsPowerShell\bin\Release\netstandard2.0\CloudEventsPowerShell.dll
  CloudEventsPowerShell -> C:\git-repos\dmilov-forks\sdk-powershell\CloudEvents.Sdk\
[11:24:10 AM] INFO: Run unit tests

Starting discovery in 9 files.
Discovery finished in 342ms.
[+] C:\git-repos\dmilov-forks\sdk-powershell\test\unit\ConvertFrom-HttpMessage.Tests.ps1 1.42s (421ms|812ms)
[+] C:\git-repos\dmilov-forks\sdk-powershell\test\unit\ConvertTo-HttpMessage.Tests.ps1 698ms (219ms|462ms)
[+] C:\git-repos\dmilov-forks\sdk-powershell\test\unit\New-CloudEvent.Tests.ps1 380ms (29ms|340ms)
[+] C:\git-repos\dmilov-forks\sdk-powershell\test\unit\Read-CloudEventData.Tests.ps1 356ms (17ms|324ms)
[+] C:\git-repos\dmilov-forks\sdk-powershell\test\unit\Read-CloudEventJsonData.Tests.ps1 384ms (44ms|327ms)
[+] C:\git-repos\dmilov-forks\sdk-powershell\test\unit\Read-CloudEventXmlData.Tests.ps1 439ms (68ms|343ms)
[+] C:\git-repos\dmilov-forks\sdk-powershell\test\unit\Set-CloudEventData.Tests.ps1 542ms (66ms|460ms)
[+] C:\git-repos\dmilov-forks\sdk-powershell\test\unit\Set-CloudEventJsonData.Tests.ps1 361ms (21ms|328ms)
[+] C:\git-repos\dmilov-forks\sdk-powershell\test\unit\Set-CloudEventXmlData.Tests.ps1 385ms (55ms|316ms)
Tests completed in 5.02s
Tests Passed: 28, Failed: 0, Skipped: 0 NotRun: 0
[11:24:17 AM] INFO: Run integration tests

Starting discovery in 1 files.
Discovery finished in 217ms.
[+] C:\git-repos\dmilov-forks\sdk-powershell\test\integration\HttpIntegration.Tests.ps1 2.84s (1.89s|757ms)
Tests completed in 2.86s
Tests Passed: 5, Failed: 0, Skipped: 0 NotRun: 0
```

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -98,4 +98,3 @@ if (Test-Path $catalogFilePath) {
    # Delete previous catalog file
    Remove-Item $catalogFilePath -Confirm:$false
 }
-New-FileCatalog -Path $OutputDir -CatalogFilePath $catalogFilePath | Out-Null
```
Comment (SimeonGerginov) [inline build.ps1:101]: Optional suggestion: Maybe we can remove the whole step **Prepare Module for Publishing** although the condition for removing the previous catalog file will never be invoked with the removal of the **NewFileCatalog** call.

**Review by SimeonGerginov — DISMISSED** [review]
LGTM!

File: src/CloudEventsPowerShell/CloudEvents.Sdk.psd1 [hunk 2]
```diff
@@ -9,7 +9,7 @@
 RootModule = 'CloudEvents.Sdk.psm1'
 
 # Version number of this module.
-ModuleVersion = '0.1.4'
+ModuleVersion = '0.2.1'
```
Comment (embano1) [inline src/CloudEventsPowerShell/CloudEvents.Sdk.psd1]: Usually we'd go with a 0.2.0, right?

File: src/CloudEventsPowerShell/CloudEvents.Sdk.psm1 [hunk 3]
```diff
@@ -72,30 +72,30 @@ PROCESS {
...
 }
 
-#region Add Data Functions
-function Add-CloudEventData {
+#region Set Data Functions
+function Set-CloudEventData {
 <#
    .SYNOPSIS
-   This function adds data to a cloud event.
+   This function sets data into a cloud event.
```
Comment (embano1) [inline src/CloudEventsPowerShell/CloudEvents.Sdk.psm1]:
```suggestion
   This function sets data in a cloud event.
```

File: src/CloudEventsPowerShell/CloudEvents.Sdk.psm1 [hunk 4]
```diff
@@ -72,30 +72,30 @@ PROCESS {
...
 
    .DESCRIPTION
-   This function adds data to a cloud event object with the provided parameters.
+   This function sets data into a cloud event object with the provided parameters.
```
Comment (embano1) [inline src/CloudEventsPowerShell/CloudEvents.Sdk.psm1]:
```suggestion
   This function sets data in a cloud event object with the provided parameters.
```

File: src/CloudEventsPowerShell/CloudEvents.Sdk.psm1 [hunk 5]
```diff
@@ -72,30 +72,30 @@ PROCESS {
...
 
    .PARAMETER CloudEvent
-   Specifies the cloud event object to add data to.
+   Specifies the cloud event object to set data into.
```
Comment (embano1) [inline src/CloudEventsPowerShell/CloudEvents.Sdk.psm1]: See above, I think it should read "to". Please also see the other places.

File: src/CloudEventsPowerShell/CloudEvents.Sdk.psm1 [hunk 6]
```diff
@@ -135,27 +135,27 @@ PROCESS {
...
+   This function converts a PowerShell hashtable to JSON format data and sets it into a cloud event.
 
    .PARAMETER CloudEvent
-   Specifies the cloud event object to add data to.
+   Specifies the cloud event object to set data into.
 
    .PARAMETER Data
-   Specifies the PowerShell hashtable object that is added as JSON to the cloud event 'data' attribute.
+   Specifies the PowerShell hashtable object that is set as JSON on the cloud event 'data' attribute.
    The 'datacontenttype' attribute is set to 'applicaiton/json'
```
Comment (embano1) [inline src/CloudEventsPowerShell/CloudEvents.Sdk.psm1]:
```suggestion
   The 'datacontenttype' attribute is set to 'application/json'
```

File: src/CloudEventsPowerShell/CloudEvents.Sdk.psm1 [hunk 7]
```diff
@@ -193,19 +193,19 @@ PROCESS {
...
+   This function converts a PowerShell hashtable to XML format data and sets it into a cloud event.
 
    .PARAMETER CloudEvent
-   Specifies the cloud event object to add data to.
+   Specifies the cloud event object to set data into.
 
    .PARAMETER Data
-   Specifies the PowerShell hashtable object that is added as XML to the cloud event 'data' attribute.
+   Specifies the PowerShell hashtable object that is set as XML on the cloud event 'data' attribute.
    The 'datacontenttype' attribute is set to 'applicaiton/xml'
```
Comment (embano1) [inline src/CloudEventsPowerShell/CloudEvents.Sdk.psm1]:
```suggestion
   The 'datacontenttype' attribute is set to 'application/xml'
```

File: test/unit/Set-CloudEventJsonData.Tests.ps1 [hunk 8]
```diff
@@ -3,9 +3,9 @@
 #  SPDX-License-Identifier: Apache-2.0
 # **************************************************************************
 
-Describe "Add-CloudEventJsonData Function Tests" {
-   Context "Adds Json Data" {
-      It 'Adds json data with depth 1' {
+Describe "Set-CloudEventJsonData Function Tests" {
+   Context "Sets Json Data" {
+      It 'Sets json data with depth 1' {
```
Comment (embano1) [inline test/unit/Set-CloudEventJsonData.Tests.ps1:8]: Looks like default depth is 1 - is this documented somewhere or general knowledge in PS community?

Comment (dmilov) [reply test/unit/Set-CloudEventJsonData.Tests.ps1:8]: No, the default is 3. Added description and of the `Depth` parameter help to address this comment. The test here covers the conversion of a hashtable with depth 1.

Comment (embano1) [reply test/unit/Set-CloudEventJsonData.Tests.ps1:8]: Thx, can we please be explicit and consistent in the test cases then too and use `-Depth n` everywhere? That is actually what threw me off during the review and why I was expecting `1` the be the default.

**Review by embano1 — APPROVED** [review]
LGTM, do you want to squash or squash during merge?

**Comment by dmilov** [comment]
> 
> 
> LGTM, do you want to squash or squash during merge?

I'll squash on merge

**Pull request merged**